### PR TITLE
Update TAs

### PR DESCRIPTION
**Pull request by dmcilvaney**
- Re-add the old authenticated variable driver
- Update the authvar driver to work with the new authvar TA
- Change the UUIDs of the TAs to avoid any conflicts with existing storage (this will reset the TPM's memory)
- Removed the old TA binaries (once the MSRsec repo is stable these will need to be replaced)
- Expand the Authvar test application
- Fix TPM hash missmatch warning (by supporting SHA1->SHA512)

### CODE REVIEW

**Comment by dmcilvaney** [comment]
Changed the target branch to avoid breaking anyone.

File: Platform/Microsoft/Application/AuthVarTest/AuthVarTest.c [hunk 1]
```diff
@@ -269,6 +283,654 @@ VerifyValidVarWriteReadEnumDel (
...
+    );
+
+  ActualDataSize = DataSize;
+  VERIFY_SUCCEEDED (gRT->GetVariable (
+                          TestVariableName2,
+                          &TestVendorGuid,
+                          &ActualVarAttributes,
+                          &ActualDataSize,
+                          mActualImageData),
+                    "Checking %s is set correctly with extra nulls",
```
Comment (neilsh-msft) [inline Platform/Microsoft/Application/AuthVarTest/AuthVarTest.c:718]: What is this actually testing?  The length of the variable name is never passed to the function, so it should always stop at the first null character.  A better test would have non-null characters following the null terminator.

File: Platform/Microsoft/Application/AuthVarTest/AuthVarTest.c [hunk 2]
```diff
@@ -137,17 +137,18 @@ VerifyVariableEnumerable (
       VERIFY_IS_NOT_NULL (mEnumVariableName);
       VariableNameBufferSize = VariableNameSize;
```
Comment (neilsh-msft) [inline Platform/Microsoft/Application/AuthVarTest/AuthVarTest.c:138]: You should include test cases for the following:
Different length variable names, what is the max length?
Different cased variable names (are they case sensitive?)
Variable names with / without accents (  a vs à )
Variable names with special characters.
etc.

File: Platform/Microsoft/Application/AuthVarTest/AuthVarTest.c (hunk 1 above)
Comment (dmcilvaney) [reply Platform/Microsoft/Application/AuthVarTest/AuthVarTest.c:718]: A lot of this testing file was build to explicitly target suspect functionality in the TA. In this case there was an inconsistency in how the length of strings was calcualted and I was trying to track that bug down.

There are deffinately holes in the test coverage, but I figured it was better than nothing. Expanding it to be more general purpose has always been on my todo list.

File: Platform/Microsoft/Application/AuthVarTest/AuthVarTest.c [hunk 3]
```diff
@@ -137,17 +137,18 @@ VerifyVariableEnumerable (
...
+                    );
+    }
+
+    if (Status == EFI_NOT_FOUND) {
       // Reached end of variable list.
       break;
     } else {
-      VERIFY_SUCCEEDED (Status, "Verify variable retrieval");
-
+      VERIFY_SUCCEEDED (Status, "Verify variable retrieval");
```
Comment (christopherco) [inline Platform/Microsoft/Application/AuthVarTest/AuthVarTest.c:151]: Nit - No trailing whitespace

```suggestion
      VERIFY_SUCCEEDED (Status, "Verify variable retrieval");
```

File: Platform/Microsoft/Application/AuthVarTest/AuthVarTest.c [hunk 4]
```diff
@@ -269,6 +283,779 @@ VerifyValidVarWriteReadEnumDel (
...
+    IN EFI_GUID *VendorGuid,
+    IN UINT32   BaseAttributes,
+    IN UINT32   ExtraSetAttributes,
+    IN UINTN    DataAddSize,
+    IN VOID     *DataAdd,
+    IN UINTN    DataCheckSize,
+    IN VOID     *DataCheck
+  )
+{
+  UINTN ActualDataSize = DataCheckSize;
```
Comment (christopherco) [inline Platform/Microsoft/Application/AuthVarTest/AuthVarTest.c:311]: Nit - EDK2 coding style mandates variable declaration and initialization to be on separate lines. Applies throughout.

File: Platform/Microsoft/Application/AuthVarTest/AuthVarTest.c [hunk 5]
```diff
@@ -269,6 +283,779 @@ VerifyValidVarWriteReadEnumDel (
...
+**/
+VOID
+DeleteAndCheckVariable(
+    IN CHAR16   *VariableName,
+    IN EFI_GUID *VendorGuid,
+    IN UINT32   BaseAttributes,
+    IN UINT32   ExtraSetAttributes
+  )
+{
+  UINTN DataSize = 0;
```
Comment (christopherco) [inline Platform/Microsoft/Application/AuthVarTest/AuthVarTest.c:364]: Nit - Variable should be declared in alphabetical order.

File: Platform/Microsoft/Application/AuthVarTest/AuthVarTest.c [hunk 6]
```diff
@@ -269,6 +283,779 @@ VerifyValidVarWriteReadEnumDel (
...
+  Buffer1 = mImageData;
+  Buffer2 = Buffer1 + ChunkSize * MaxChunksPerVar;
+  Buffer3 = Buffer2 + ChunkSize * MaxChunksPerVar;
+
+  mActualImageData = AllocatePool(BufferSize);
+  VERIFY_IS_NOT_NULL(mActualImageData);
+
+  // To allow for checking persistance it is important that the data
+  // is deterministic.
+  for(Idx = 0; Idx < BufferSize; Idx++) {
```
Comment (christopherco) [inline Platform/Microsoft/Application/AuthVarTest/AuthVarTest.c:445]: Nit - Space after for
```suggestion
  for (Idx = 0; Idx < BufferSize; Idx++) {
```

File: Platform/Microsoft/Application/AuthVarTest/AuthVarTest.c [hunk 7]
```diff
@@ -269,6 +283,779 @@ VerifyValidVarWriteReadEnumDel (
...
+  STATIC CHAR16 QueryVariableName[] = L"QueryTestVariable";
+  UINT64 MaximumVariableStorageSize;
+  UINT64 RemainingVariableStorageSize;
+  UINT64 MaximumVariableSize;
+  UINT64 LargerVariableSize;
+  UINT64 InitialSpaceAvailable;
+  UINT64 InitialMaximumVariableSize;
+  UINT32 i;
+
+  for(i = 0; i < 2; i++) {
```
Comment (christopherco) [inline Platform/Microsoft/Application/AuthVarTest/AuthVarTest.c:871]: Nit - space
```suggestion
  for (i = 0; i < 2; i++) {
```

File: Platform/Microsoft/Application/AuthVarTest/AuthVarTest.c [hunk 8]
```diff
@@ -269,6 +283,779 @@ VerifyValidVarWriteReadEnumDel (
...
+                      );
+
+    VERIFY_IS_TRUE (RemainingVariableStorageSize <= (InitialSpaceAvailable - LargerVariableSize),
+                    "0x%llx bytes were available, stored 0x%llx bytes, now 0x%llx available",
+                    InitialSpaceAvailable,
+                    LargerVariableSize,
+                    RemainingVariableStorageSize
+                    );
+
+  /*  VERIFY_IS_GREATER_THAN (UINT64,
```
Comment (christopherco) [inline Platform/Microsoft/Application/AuthVarTest/AuthVarTest.c:918]: Nit - no commented out code upstream. Applies throughout

File: Platform/Microsoft/OpteeClientPkg/Drivers/AuthVarOpteeRuntimeDxe/AuthVarsDxe.c [hunk 9]
```diff
@@ -0,0 +1,1589 @@
...
+  IN UINT32                VariableResultSize,
+  IN TEEC_SharedMemory     *VariableResult,
+  OUT UINT32               *ResultSize,
+  OUT UINT32               *AuthVarStatus
+)
+{
+  EFI_STATUS Status = EFI_SUCCESS;
+  TEEC_Operation TeecOperation = { 0 };
+  TEEC_Result TeecResult = TEEC_SUCCESS;
+  uint32_t ErrorOrigin = 0;
```
Comment (christopherco) [inline Platform/Microsoft/OpteeClientPkg/Drivers/AuthVarOpteeRuntimeDxe/AuthVarsDxe.c:213]: nit - use EDK2 types

File: Platform/Microsoft/OpteeClientPkg/Drivers/AuthVarOpteeRuntimeDxe/AuthVarsDxe.c [hunk 10]
```diff
@@ -0,0 +1,1589 @@
...
+    LOG_ERROR(
+      "CreateEventEx(gEfiEventExitBootServicesGuid) failed. (Status=%r)",
+      Status);
+
+    goto Exit;
+  }
+
+Exit:
+  return Status;
+}
```
Comment (christopherco) [inline Platform/Microsoft/OpteeClientPkg/Drivers/AuthVarOpteeRuntimeDxe/AuthVarsDxe.c:1589]: Nit - end files with newline

File: Platform/Microsoft/OpteeClientPkg/Drivers/AuthVarOpteeRuntimeDxe/Measurement.c [hunk 11]
```diff
@@ -0,0 +1,252 @@
+/** @file
+Measure TrEE required variable.
+Copyright (c) 2013 - 2014, Intel Corporation. All rights reserved.<BR>
```
Comment (christopherco) [inline Platform/Microsoft/OpteeClientPkg/Drivers/AuthVarOpteeRuntimeDxe/Measurement.c:3]: Need to add our copyright

File: Platform/Microsoft/OpteeClientPkg/Drivers/AuthVarOpteeRuntimeDxe/Measurement.c [hunk 12]
```diff
@@ -0,0 +1,252 @@
...
+@param[in]  VariableName      A Null-terminated string that is the name of the vendor's variable.
+@param[in]  VendorGuid        A unique identifier for the vendor.
+@retval TRUE                  This is SecureBootPolicy Variable
+@retval FALSE                 This is not SecureBootPolicy Variable
+**/
+BOOLEAN
+IsSecureBootPolicyVariable(
+  IN CHAR16                                 *VariableName,
+  IN EFI_GUID                               *VendorGuid
+)
```
Comment (christopherco) [inline Platform/Microsoft/OpteeClientPkg/Drivers/AuthVarOpteeRuntimeDxe/Measurement.c:50]: nit - needs 2 spaces

```suggestion
  )
```

File: Platform/Microsoft/OpteeClientPkg/Drivers/AuthVarOpteeRuntimeDxe/UEFIVarServices.h [hunk 13]
```diff
@@ -0,0 +1,146 @@
+/*++
+Copyright (c) 2013 Microsoft Corporation. All Rights Reserved.
```
Comment (christopherco) [inline Platform/Microsoft/OpteeClientPkg/Drivers/AuthVarOpteeRuntimeDxe/UEFIVarServices.h:2]: Need to fix this header comment. Add license

File: Platform/Microsoft/OpteeClientPkg/Drivers/AuthVarOpteeRuntimeDxe/UEFIVarServices.h [hunk 14]
```diff
@@ -0,0 +1,146 @@
...
+typedef struct _VS_GET_FRAG_STATS
+{
+    UINT32 NonVolatileSpaceUsed;
+    UINT32 IdealNonVolatileSpaceUsed;
+    UINT32 IdealWithOverheadSpace;
+} VS_GET_FRAG_STATS, *PVS_GET_FRAG_STATS;
+
+
+//
+// ISSUE-2013/06/14-ARNOLDP: Refactor Ntefi.h
```
Comment (christopherco) [inline Platform/Microsoft/OpteeClientPkg/Drivers/AuthVarOpteeRuntimeDxe/UEFIVarServices.h:107]: Remove

**Review by christopherco — APPROVED** [review]
Generally looks good. Nits can be ignored if we're going to do a codestyle pass before it goes into master.

**Pull request merged**